agent/engine/service_connect/manager_linux.go

```diff
@@ -150,10 +156,24 @@ func (m *manager) initAgentDirectoryMounts(taskId string, container *apicontaine
 
 func (m *manager) initAgentEnvironment(container *apicontainer.Container) {
 	scEnv := map[string]string{
-		m.relayENV:   unixRequestPrefix + filepath.Join(m.relayPathContainer, m.relayFileName),
-		m.statusENV:  filepath.Join(m.statusPathContainer, m.statusFileName),
-		agentModeENV: agentModeValue,
-		agentAuthENV: agentAuthOff,
+		m.endpointENV: unixRequestPrefix + filepath.Join(m.relayPathContainer, m.relayFileName),
+		m.statusENV:   filepath.Join(m.statusPathContainer, m.statusFileName),
+		agentModeENV:  agentModeValue,
+		agentAuthENV:  agentAuthOff,
+	}
+
+	container.MergeEnvironmentVariables(scEnv)
+}
+
+func (m *manager) initRelayEnvironment(config *config.Config, container *apicontainer.Container) {
+	scEnv := map[string]string{
+		m.statusENV:    filepath.Join(m.statusPathContainer, m.statusFileName),
+		upstreamENV:    filepath.Join(m.relayPathContainer, m.relayFileName),
+		regionENV:      config.AWSRegion,
+		agentModeENV:   agentModeValue,
+		relayEnableENV: relayEnableOn,
+		// TODO: get programmatically when available
```

Comment:
This actually is already available through `DiscoverPollEndpoint`. We currently are only consuming ACS and TACS endpoints, need to make a small [change](https://github.com/aws/amazon-ecs-agent/blob/71a0f6765cc469913a0b09728196feb0d0d33e37/agent/ecs_client/model/ecs/api.go#L7401) to take in EMS endpoint as well

Reply:
Need to update API model [here](https://github.com/aws/amazon-ecs-agent/blob/feature/service-connect/agent/ecs_client/model/api/api-2.json#L1147-L1153). We can do that in a follow-up PR.